### PR TITLE
added ededup python implementation

### PR DESCRIPTION
**Pull request by blublinsky**

### CODE REVIEW

File: transforms/universal/ededup/python/Makefile [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Define the root of the local git clone for the common rules to be able 
+# know where they are running from.
+REPOROOT=../../../..
+# Include a library of common .transform.* targets which most
+# transforms should be able to reuse.  However, feel free
+# to override/redefine the rules below. 
+
+# $(REPOROOT)/.make.versions file contains the versions
+
+TRANSFORM_RUNTIME=python
```
Comment (daw3rd) [inline transforms/universal/ededup/python/Makefile]: TRANFORM_RUNTIME should not be necessary as it should default to the name of the containing directory (i.e. python).  are you finding it required?

File: transforms/universal/ededup/python/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+## Summary 
+This is a python version of ededup
+
+## Configuration and command line Options
+
+The set of dictionary keys holding [EdedupPythonTransform](src/ededup_transform_python.py) 
+configuration for values are as follows:
+
+* _ededup_doc_column_ - specifies the name of the column containing a document
+the `transform()` method of `EdedupPythonTransform`.
```
Comment (daw3rd) [inline transforms/universal/ededup/python/README.md]: seems like something got dropped here

File: transforms/universal/ededup/python/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## Running
+
+### Launched Command Line Options 
+The following command line arguments are available in addition to 
+the options provided by 
+the [python launcher](../../../../data-processing-lib/doc/python-launcher-options.md).
+```
+ --ededup_doc_column EDEDUP_DOC_COLUMN
+                        name of the column containing documant
```
Comment (daw3rd) [inline transforms/universal/ededup/python/README.md]: documant -> document

File: transforms/universal/ededup/python/src/ededup_transform_base.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    """
+
+    def __init__(self, config: dict):
+        """
+        Initialize based on the dictionary of configuration information.
+        The dictionary should contain the following:
+            doc_column - name of the doc column
+        """
+        super().__init__(config)
+        self.doc_column = config.get("doc_column", "")
```
Comment (daw3rd) [inline transforms/universal/ededup/python/src/ededup_transform_base.py]: This needs a default value

File: transforms/universal/ededup/python/Makefile (hunk 1 above)
Comment (blublinsky) [reply transforms/universal/ededup/python/Makefile]: Its really a mess at the moment. This is cut and paste from NOOP. I looked at several of them and they are slightly different

**Review by daw3rd — CHANGES_REQUESTED** [review]

File: transforms/universal/ededup/python/README.md (hunk 2 above)
Comment (blublinsky) [reply transforms/universal/ededup/python/README.md]: fixed

File: transforms/universal/ededup/python/README.md (hunk 3 above)
Comment (blublinsky) [reply transforms/universal/ededup/python/README.md]: fixed

File: transforms/universal/ededup/python/src/ededup_transform_base.py (hunk 4 above)
Comment (blublinsky) [reply transforms/universal/ededup/python/src/ededup_transform_base.py]: fixed

File: transforms/universal/ededup/python/README.md (hunk 2 above)
Comment (daw3rd) [reply transforms/universal/ededup/python/README.md]: Still unreadable.  Suggest
`
specifies the name of the column containing a document that
the `transform()` method of `EdedupPythonTransform` examines for duplicates 
among other documents. `

File: transforms/universal/ededup/python/src/ededup_transform_base.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    """
+
+    def __init__(self, config: dict):
+        """
+        Initialize based on the dictionary of configuration information.
+        The dictionary should contain the following:
+            doc_column - name of the doc column
+        """
+        super().__init__(config)
+        self.doc_column = config.get("doc_column", "contents")
```
Comment (daw3rd) [inline transforms/universal/ededup/python/src/ededup_transform_base.py:72]: would be better to define global constants for these and other strings, similar to the way many other transforms do it.

Comment (blublinsky) [reply transforms/universal/ededup/python/src/ededup_transform_base.py:72]: its a matter of preference

**Review by daw3rd — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -63,25 +63,25 @@ Support for additional data modalities is expected in the future and additional
...
+|Doc ID annotation               | :white_check_mark: |:white_check_mark:|:white_check_mark:|:white_check_mark:      |
+|Programming language annnotation| :white_check_mark: |:white_check_mark:|                  |:white_check_mark:      | 
+|Exact dedup filter              | :white_check_mark: |:white_check_mark:|                  |:white_check_mark:      |
+|Fuzzy dedup filter              |                    |:white_check_mark:|                  |:white_check_mark:      |
+|Code quality annotation         | :white_check_mark: |:white_check_mark:|                  |:white_check_mark:      |
+|Malware annotation              | :white_check_mark: |:white_check_mark:|                  |:white_check_mark:      |
+|Filter on annotations           | :white_check_mark: |:white_check_mark:|:white_check_mark:|:white_check_mark:      |
+|Language identification         | :white_check_mark: |:white_check_mark:|                  |:white_check_mark:      |
+|Code (from zip) to Parquet      | :white_check_mark: |:white_check_mark:|                  |:white_check_mark:      |
```
Comment (daw3rd) [inline README.md]: Doc quality and PDF to parquet got dropped

**Review by daw3rd — CHANGES_REQUESTED** [review]

**Review by daw3rd — APPROVED** [review]

**Pull request merged**